### ISSUE TITLE
Package is now broken for imports

### ISSUE THREAD

**Issue by niktek** [bobthered/tailwindcss-palette-generator#3]
Hi,

I think things might have broken in the package.json when you did: https://github.com/bobthered/tailwindcss-palette-generator/commit/a28edfee3332772e347a6fdc66b13ee0419ef239

We are now getting:
`Failed to resolve entry for package "@bobthered/tailwindcss-palette-generator". The package may have incorrect main/module/exports specified in its package.json.`

Cheers,
Nik

**Comment by bobthered** [comment]
Nik,
The recent change adding typescript added a breaking change.  Without seeing your code, I believe the issue is your import statement.  See the docs as you need to import like below:

```js
import { tailwindcssPaletteGenerator } from '@bobthered/tailwindcss-palette-generator';

// or
const { tailwindcssPaletteGenerator } = require('@bobthered/tailwindcssPaletteGenerator');
```

I'll look into providing a default export to fix the issue.

**Comment by bobthered** [comment]
This is my first time working with typescript, so I'm still figuring it out.  I'm not sure how to serve both a cjs and esm file with default and named exports.  Any idea on this one?  I'm familiar with esm to serve both of these, but can't figure it out with cjs.  After reading a little, it looks like you can't use module.exports... & exports... in the same file.  Is that correct?  I'd like something like the following

```js
// dist/cjs/index.js
module.exports = tailwindcssPaletteGenerator;
exports.tailwindcssPaletteGenerator = tailwindcssPaletteGenerator;

// dist/esm/index.js
export default tailwindcssPaletteGenerator;
export { tailwindcssPaletteGenerator };
```

**Comment by niktek** [comment]
Hi,

Looks like you need to update your package.json so that "module": "./dist/esm/index.js" is there instead.

And I feel your pain! Just been through a whole bunch of this stuff in creating the CLI for [Skeleton](http://skeleton.dev) with TS, and it's still not 100% there.  Can also recommend looking into [tsup](https://tsup.egoist.dev) for helping with the bundling - though, if you have it working with `tsc`, then there is an argument to keep it simple.  Might also want to add `dist` to your .gitignore as it is just an output folder and doesn't need to mess with your git history by having commits related to it.

Cheers,
Nik

**Comment by bobthered** [comment]
I fixed the package.json module path and published the update ( [3.1.1](https://www.npmjs.com/package/@bobthered/tailwindcss-palette-generator) ).

I am still at a loss on how to provide default and named exports for cjs from typescript.  Adding typescript, I've introduced a breaking change since the import statement has changed:

```diff
- const tailwindcssPaletteGenerator = require('@bobthered/tailwindcss-palette-generator');
+ const { tailwindcssPaletteGenerator } = require('@bobthered/tailwindcss-palette-generator');
```

I believe the esm is working as intended with both default and named imports available:
```js
import generatePalette from '@bobthered/tailwindcss-palette-generator';
```
or
```js
import { tailwindcssPaletteGenerator } from '@bobthered/tailwindcss-palette-generator';
```
